Kernel/FileSystem/FUSE/FileSystem.cpp

```diff
@@ -70,9 +70,22 @@ Inode& FUSE::root_inode()
     return *m_root_inode;
 }
 
-ErrorOr<void> FUSE::rename(Inode&, StringView, Inode&, StringView)
+ErrorOr<void> FUSE::rename(Inode& old_parent_inode, StringView old_basename, Inode& new_parent_inode, StringView new_basename)
 {
-    return ENOTIMPL;
+    auto payload = TRY(KBuffer::try_create_with_size("FUSE: Lookup name string"sv, sizeof(fuse_rename_in) + old_basename.length() + new_basename.length() + 2));
+    memset(payload->data(), 0, payload->size());
+    fuse_rename_in* rename_in = bit_cast<fuse_rename_in*>(payload->data());
+    rename_in->newdir = new_parent_inode.identifier().index().value();
+    memcpy(payload->data() + sizeof(fuse_rename_in), old_basename.characters_without_null_termination(), old_basename.length());
+    memcpy(payload->data() + sizeof(fuse_rename_in) + old_basename.length() + 1, new_basename.characters_without_null_termination(), new_basename.length());
```

Comment:
urgh, ideally these would have a trailing array member to memcpy to

Reply:
Well in this case the boon isn't as big as with the others, but it's a common pattern with the packet like types we have:
```c++
struct fuse_rename_in {
    uint64_t newdir;
    uint8_t data[];
};

/* ... */

memcpy(rename_in.data, old_basename.characters_without_null_termination(), old_basename.length());
memcpy(rename_in.data + old_basename.length() + 1 , new_basename.characters_without_null_termination(), new_basename.length());
```